serializations.md

```diff
@@ -480,3 +481,23 @@ The channel offchain transaction is not included directly in the transaction tre
 , <closes_at>       , int()
 ]
 ```
+
+#### Proof of inclusion on state trees (POI)
+```
+[ {<accounts>  , [{binary(), [{binary(), [binary()]}]}]}
+, {<calls>     , [{binary(), [{binary(), [binary()]}]}]}
+, {<channels>  , [{binary(), [{binary(), [binary()]}]}]}
+, {<contracts> , [{binary(), [{binary(), [binary()]}]}]}
+, {<ns>        , [{binary(), [{binary(), [binary()]}]}]}
+, {<oracles>   , [{binary(), [{binary(), [binary()]}]}]}
+]
+```
+
+Each field is the representation of a subtree to the top level tree. The serialization should be interpreted as
+```
+{<tree>, [<proof_of_inclusion>]}
+where
+<proof_of_inclusion> := [{<root_hash>, [{<mpt_hash>, <mpt_value>}]}]
+```
+where `[{<mpt_hash>, <mpt_value>}]` is the sorted list of Merkel Patricia Tree nodes in the proof. If the subtree is empty,
+the serialization is just `[]`, and otherwise it is a list of one element `[<proof_of_inclusion>]`.
```

Comment:
I understand this sentence is clarifying length of list `[<proof_of_inclusion>]` in `{<tree>, [<proof_of_inclusion>]}`.

A. Just confirming: `[]` is not a proof of absence.

B. I understand that "otherwise it is a list of one element `[<proof_of_inclusion>]`" means that this serialization can carry at most one value per subtree e.g.: one account; or one account plus one channel.

C. This implies that a POI can be serialized with no value in any subtree. Useless but legit - ok.

D. I see absent the hash of the top state tree (including accounts, channels, etc.) i.e. the one that is in block header. I think this is not sufficient, as this requires user to have not only the block headers but also the root hash of each subtree (accounts, channels, etc.) for each header.

Reply:
A. This is correct.

B. No. The proof of inclusion for one tree can carry any amount of objects in the tree. Basically, the proof is a pruned node db with enough information for any number of paths in the tree.

C. Yes.

D. The root hash (from the header) can be [calculated](https://github.com/aeternity/epoch/pull/1202/files#diff-29b0252c569ea38faaa5fed4b9820676R301) from the root hashes of the subtrees. The actual root hash is omitted on purpose since it can be calculated. Analogous with the blocks (and headers) not having their own hashes in the representation.
The proof of inclusion always contains all subtrees, since this is needed to verify the root hash.

Reply:
>  The proof of inclusion always contains all subtrees, since this is needed to verify the root hash.

This is the only bit unclear to me.